### ISSUE TITLE
extmod: duplicate symbols when shared linking

### ISSUE THREAD

**Issue by pmp-p** [micropython/micropython#5018]
wasm-ld of clang 10 does not allow linking STATIC defined as static for void, or most likely functions pointer addresses would get corrupted ( this is actually the case in standard emsdk fastcomp branch https://github.com/emscripten-core/emscripten/issues/8268  )

but redefining STATIC to nothing leads to some collisions:

```
wasm-ld: error: duplicate symbol: heap_siftdown
>>> defined in build/extmod/moduheapq.o
>>> defined in build/extmod/modutimeq.o

wasm-ld: error: duplicate symbol: heap_siftup
>>> defined in build/extmod/moduheapq.o
>>> defined in build/extmod/modutimeq.o
```

(heap_siftdown and heap_siftup seem identical ), maybe get_heap need special treatment.

**Comment by dpgeorge** [comment]
Thanks for the report.  There are indeed a few of these cases.

Since it was intended to be able to compile with `STATIC` defined to nothing these cases should all be fixed.

**Comment by dpgeorge** [comment]
Fixed by 0cc8910bc58c1a1f69a696f5efe4ccf0ffcf2984
